src/components/views/rooms/RoomDetailList.js

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2017 New Vector Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import sdk from '../../../index';
+import dis from '../../../dispatcher';
+import React from 'react';
+import { _t } from '../../../languageHandler';
+import linkifyString from 'linkifyjs/string';
+import sanitizeHtml from 'sanitize-html';
+import { ContentRepo } from 'matrix-js-sdk';
+import MatrixClientPeg from '../../../MatrixClientPeg';
+import PropTypes from 'prop-types';
+
+function getDisplayAliasForRoom(room) {
+    return room.canonical_alias || (room.aliases ? room.aliases[0] : "");
+}
+
+const RoomDetailRow = React.createClass({
+    onClick: function(ev) {
+        ev.preventDefault();
+        dis.dispatch({
+            action: 'view_room',
+            room_id: this.props.room.room_id,
+        });
+    },
+
+    render: function() {
+        const BaseAvatar = sdk.getComponent('avatars.BaseAvatar');
+
+        const room = this.props.room;
+        const name = room.name || getDisplayAliasForRoom(room) || _t('Unnamed room');
+        const topic = linkifyString(sanitizeHtml(room.topic || ''));
+
+        const guestRead = room.world_readable ? (
+                <div className="mx_RoomDirectory_perm">{ _t('World readable') }</div>
```

Comment:
I guess this should share code with the public room directory, at least in the long term

Reply:
Sure, in the long term. It was too entrenched in RoomDirectory so I decided not to do that yet.